test/sql/simple/test_like_escape.cpp

```diff
@@ -0,0 +1,68 @@
+#include "catch.hpp"
+#include "test_helpers.hpp"
+
+using namespace duckdb;
+using namespace std;
+
+TEST_CASE("Test scalar LIKE statement with custom ESCAPE", "[like]") {
+	unique_ptr<QueryResult> result;
+	DuckDB db(nullptr);
+	Connection con(db);
+
+	// scalar like with escape
+	result = con.Query("SELECT '%++' LIKE '*%++' ESCAPE '*';");
+	REQUIRE(CHECK_COLUMN(result, 0, {Value::BOOLEAN(true)}));
+	// Not Like
+	result = con.Query("SELECT '%++' NOT LIKE '*%++' ESCAPE '*';");
+	REQUIRE(CHECK_COLUMN(result, 0, {Value::BOOLEAN(false)}));
+	// Default tests
+	result = con.Query("SELECT '\' LIKE '\\' ESCAPE '\';");
```

Comment:
Take care to properly escape backslashes here, as the queries are inside C-style strings. All the backslashes should be doubled. This should be:

`SELECT '\\' LIKE '\\\\' ESCAPE '\\';`